Adding Unifi USG instructions (#730)
pi-hole/docs: docs/routers/ubiquiti-usg.md

This guide was developed using a Ubiquiti Cloud Key v2 controller running v7.1.61 firmware. Following these steps will have you adding your Pi-hole as a DNS server for your IP stack for all devices on your LAN.
>>>>>>> f5fcb0a (Updated ToC and router page)
=======
This guide was developed using a Ubiquiti Cloud Key v2 controller running v7.1.61 firmware. Following the first method below will have you adding your Pi-hole as a DNS server for your IP stack for all devices on your LAN. Following the second method below will have DNS queries route through your USG and then to your Pi-hole. Either option is valid, depending on your specific requirements, but it is recommended to choose one or the other - not both.
>>>>>>> 262344b (Made changes per request)
=======
This guide was developed using a Ubiquiti Cloud Key v2 with UniFi Network v7.1.61. Following the first method below will have you adding your Pi-hole as a DNS server for all devices on your LAN. Following the second method below will have DNS queries route through your USG and then to your Pi-hole. Either option is valid, depending on your specific requirements, but it is recommended to choose one or the other - not both.

> >>>>>>> c05eb7d (Update docs/routers/ubiquiti-usg.md)

Reviewer: You have merge conflicts here and below. (The whole file shows conflicts)

This guide was developed using a Ubiquiti Cloud Key v2 controller running v7.1.61 firmware. Following these steps will have you adding your Pi-hole as a DNS server for your IP stack for all devices on your LAN.

> Note:
While this guide attempts to cover both IPv4 and IPv6, your mileage may vary.

> ### Grab your v4 and v6 address from your Raspi

Reviewer 2: ```suggestion
### Grab your IPv4 and IPv6 address from your Raspi
```

> SSH into your Raspi. A quick `ifconfig eth0` (substituting `eth0` for whatever network port on your Pi you're working with) will get you the address information for your Ethernet port:

Reviewer 2: ```suggestion
SSH into your Raspi. A quick `ifconfig eth0` (substituting `eth0` for your main interface) will get you the address information:
```

```
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.9  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 2001:470:5:3::8ed  prefixlen 128  scopeid 0x0<global>
        inet6 2001:470:5:3:8d70:e92e:7030:6eef  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::337f:710d:6781:4c85  prefixlen 64  scopeid 0x20<link>
        ether dc:a6:32:0e:31:2d  txqueuelen 1000  (Ethernet)
        RX packets 1297459  bytes 1279521260 (1.1 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2315988  bytes 2766370095 (2.5 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

> Take note of the `inet` address, and the `inet6` address which starts with `fe` (denotes a local link or LAN address). Those are the IPs which you'll use as your DNS servers in the USG. For this example, we'll be using the `192.168.0.9` and `fe80::337f:710d:6781:4c85` IPs for the following directions

Reviewer 2: ```suggestion
Take note of the `inet` address, and the `inet6` address which starts with `fe` (denotes a local link or LAN address). Those are the IPs which you'll use as your DNS servers in the USG. For this example, we'll be using the `192.168.0.9` and `fe80::337f:710d:6781:4c85` IPs for the following directions.
```

### Distribute Pi-hole as DNS server via DHCP

Go to **Settings** --> **Networks** and click on the Network line that you want to modify:

![Screenshot of USG LAN](../images/usg-lan.png)

Under **DHCP**, in the DHCP Server Management section click "Show options" to reveal the **DHCP DNS Server** section. Check the **Enable** box and enter your Raspi's IPv4 address here.

![Screenshot of USG LAN DHCP settings](../images/usg-lan-dhcp.png)

Scroll down to **IPv6**, and under the **DHCPv6/RDNSS DNS Control** section, uncheck **Auto** if it's enabled and enter Raspi's IPv6 address here. Click **Apply Changes** to have the settings persistently saved.

![Screenshot of USG LAN DHCPv6 settings](../images/usg-lan-dhcpv6.png)

All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution in the IP stack. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.

### Optional Step - Using your Pi-hole as a resolve for your USG

> Completing this *optional* step, Pi-hole can be used by your USG as an upstream DNS server, with its DNS queries taking the following path:

Reviewer 2: ..only if the step above has not been executed. If Pi-hole's IP is distributed via DHCP, queries will go there.

```
(Clients) -> USG -> Pi-hole -> Upstream DNS Server
```

> You may want to do this for a variety of use cases. For example, you may want to have exclude a development or other test environment excluded from your normal IP stack.

Reviewer 2: ```suggestion
You may want to do this for a variety of use cases. For example, you may want to exclude a development or other test environment from your normal IP stack.
```

> you prefer any DNS queries coming directly from your USG to use Pi-hole. In my case, I wanted to enable this so I'm certain 100% of my network's DNS queries were forwarded using [DNS-over-HTTPS](https://docs.pi-hole.net/guides/dns/cloudflared/).

Reviewer 2: This sentence has no beginning.

!!! warning

>     Keep in mind that if you are using your USG's IP address as a resolver for your Pi-hole, ***this will cause a resolver loop***. Make sure to only have true upstream DNS servers set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).

Reviewer 2: ```suggestion
    Keep in mind that if you are using your USG's IP address as an upstream resolver for your Pi-hole together with this configuration, ***this will cause a resolver loop***. Make sure you have an other upstream DNS servers set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).
```

> This guide was developed using a Ubiquiti Cloud Key v2 controller running v7.1.61 firmware. Following the first method below will have you adding your Pi-hole as a DNS server for your IP stack for all devices on your LAN. Following the second method below will have DNS queries route through your USG and then to your Pi-hole. Either option is valid, depending on your specific requirements, but it is recommended to choose one or the other - not both.

Reviewer 2: ```suggestion
This guide was developed using a Ubiquiti Cloud Key v2 with UniFi Network v7.1.61. Following the first method below will have you adding your Pi-hole as a DNS server for all devices on your LAN. Following the second method below will have DNS queries route through your USG and then to your Pi-hole. Either option is valid, depending on your specific requirements, but it is recommended to choose one or the other - not both.
```

> Note:
While this guide attempts to cover both IPv4 and IPv6, your mileage may vary.

### Grab your IPv4 and IPv6 address from your Raspi

> SSH into your Raspi. A quick `ifconfig eth0` (substituting `eth0` for whatever network port on your Pi you're working with) will get you the address information:

Reviewer 2: ```suggestion
SSH into your Raspi. A quick `ifconfig eth0` (substituting `eth0` for the applicable network interface) will get you the address information:
```

```

Take note of the `inet` address, and the `inet6` address which starts with `fe` (denotes a local link or LAN address). Those are the IPs which you'll use as your DNS servers in the USG. For this example, we'll be using the `192.168.0.9` and `fe80::337f:710d:6781:4c85` IPs for the following directions.

### Method 1: Distribute Pi-hole as DNS server via DHCP

Choosing this option will have DNS queries taking the following path:

```
(Clients) -> Pi-hole -> Upstream DNS Server
```

Go to **Settings** --> **Networks** and click on the Network line that you want to modify:

![Screenshot of USG LAN](../images/usg-lan.png)

Under **DHCP**, in the DHCP Server Management section click "Show options" to reveal the **DHCP DNS Server** section. Check the **Enable** box and enter your Raspi's IPv4 address here.

![Screenshot of USG LAN DHCP settings](../images/usg-lan-dhcp.png)

Scroll down to **IPv6**, and under the **DHCPv6/RDNSS DNS Control** section, uncheck **Auto** if it's enabled and enter Raspi's IPv6 address here. Click **Apply Changes** to have the settings persistently saved.

![Screenshot of USG LAN DHCPv6 settings](../images/usg-lan-dhcpv6.png)

> All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution in the IP stack. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.

Reviewer 2: ```suggestion
All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.
```

### Method 2: Using your Pi-hole as a resolve for your USG

> If you'd rather have your USG forward all DNS traffic to Pi-hole, you can complete this *optional* step. Doing so will have DNS queries taking the following path:

Reviewer 2: ```suggestion
If you'd rather have your USG forward all DNS traffic to Pi-hole, you can complete this *alternative* step. Doing so will have DNS queries taking the following path:
```

![Screenshot of USG LAN](../images/usg-lan.png)

Under **DHCP**, in the DHCP Server Management section click "Show options" to reveal the **DHCP DNS Server** section. Check the **Enable** box and enter your Raspi's IPv4 address here.

![Screenshot of USG LAN DHCP settings](../images/usg-lan-dhcp.png)

Scroll down to **IPv6**, and under the **DHCPv6/RDNSS DNS Control** section, uncheck **Auto** if it's enabled and enter Raspi's IPv6 address here. Click **Apply Changes** to have the settings persistently saved.

![Screenshot of USG LAN DHCPv6 settings](../images/usg-lan-dhcpv6.png)

All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution in the IP stack. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.

### Method 2: Using your Pi-hole as a resolve for your USG

If you'd rather have your USG forward all DNS traffic to Pi-hole, you can complete this *optional* step. Doing so will have DNS queries taking the following path:

```
(Clients) -> USG -> Pi-hole -> Upstream DNS Server
```

> You may want to do this for a variety of use cases. For example, you may want to have exclude a development or other test environment excluded from your normal IP stack. In my case, I wanted to enable this so I would be certain 100% of my network's DNS queries were forwarded using [DNS-over-HTTPS](https://docs.pi-hole.net/guides/dns/cloudflared/) and so I could get the benefit of individual client reporting via Pi-hole's admin interface.

Reviewer 2: > I could get the benefit of individual client reporting via Pi-hole's admin interface.

How does this work if all queries are send to your USG? They will appear to Pi-hole as if they come only from this device.

Reviewer 2: @Falc0n2k 

`(Clients) -> USG -> Pi-hole -> Upstream DNS Server`
won't show

> individual client reporting via Pi-hole's admin interface
